Update changelog for beta.20 (#351)
mapbox/mapbox-maps-android: CHANGELOG.md

Mapbox welcomes participation and contributions from everyone.

# 10.0.0-beta.20 - May 20, 2021
## Breaking changes ⚠️
* Introduce ResourceOptionsManager to configure the default resource options, and removed the xml configuration options for cache path and tile store path. ([#339](https://github.com/mapbox/mapbox-maps-android/pull/339))
* Rename ambient cache database to ambient_cache.db  ([#314](https://github.com/mapbox/mapbox-maps-android/pull/314))
* Remove usage of asset path from the codebase, as it is not used in Android SDK. ([#334](https://github.com/mapbox/mapbox-maps-android/pull/334))
* CameraManager#setBounds return type changed from void to expected
* Expose tileset version for sources that use TileJSON metadata
* ResourceRequest offline-data boolean field is replaced with the source string field, which whether the response came from network, cache or tile store

> * Remove GeoJSON specific methods from 'StyleManager'. Map#queryFeatureExtensions provides the same functionality

Reviewer: ```suggestion
* Remove GeoJSON specific methods from StyleManager. Map#queryFeatureExtensions provides the same functionality
```

      * Otherwise method will be synchronous.
      *
      * @param value the feature collection
      * @param onDataParsed optional callback notifying when data is parsed on a worker thread
      */
     fun featureCollection(
       value: FeatureCollection,
       onDataParsed: ((GeoJsonSource) -> Unit)? = null
     )
   ```
   Similar functions are introduced for `feature` and `geometry`.

## Features ✨ and improvements 🏁
* Add a `cameraOptions(cameraState, builderBlock)` inline method that helps mutate an existing `CameraState` object. ([#317](https://github.com/mapbox/mapbox-maps-android/pull/317))
* Add selected state handling to annotation plugin ([#316](https://github.com/mapbox/mapbox-maps-android/pull/316))
* Add API for disabling vertical/horizontal scroll gestures ([#319](https://github.com/mapbox/mapbox-maps-android/pull/319))
* Introduce API to enable render cache feature that could bring up rendering performance improvement. ([#326](https://github.com/mapbox/mapbox-maps-android/pull/326))
* Add `removeAnnotationManager` API. ([#330](https://github.com/mapbox/mapbox-maps-android/pull/330))
* Improve terrain's rendering performance 

> * Set 'begin' and 'end' timestamps for StyleLoaded and MapLoaded events, so that developers could check how much time it takes to load style and map, respectively

Reviewer: ```suggestion
* Set `begin` and `end` timestamps for StyleLoaded and MapLoaded events, so that developers could check how much time it takes to load style and map, respectively
```

> * Added "tile-requests-delay" and "tile-network-requests-delay" runtime source properties - an API for tile requests delay

Reviewer: ```suggestion
* Added `tile-requests-delay` and `tile-network-requests-delay` runtime source properties - an API for tile requests delay
```

* Introduce MapOptions.optimizeForTerrain option that allow style rendering optimizations for terrain rendering

> * The 'text-line-height' is now data-driven property

Reviewer: ```suggestion
* The `text-line-height` is now data-driven property
```

* MapLoaded, StyleLoaded and StyleDataLoaded events now contain begin and end timestamps reflecting the effective duration timespan

> * Implement geojson source's generateId feature. When enabled, the feature.id property will be auto assigned based on its index in the features array

Reviewer: Ticketed at https://github.com/mapbox/mapbox-maps-android/issues/353

> * Implement geojson source's generateId feature. When enabled, the feature.id property will be auto assigned based on its index in the features array

Reviewer: ```suggestion
```

## Bug fixes 🐞
* Fix scalebar doesn't refresh issue. ([#331](https://github.com/mapbox/mapbox-maps-android/pull/331))
* Trigger nested high-level animator listener correctly. ([#335](https://github.com/mapbox/mapbox-maps-android/pull/335))
* Make compass visible when camera was mutated while compass was disabled. ([#322](https://github.com/mapbox/mapbox-maps-android/pull/322))
* Enable LocationComponent automatically when style loaded; fix null island location puck ([#333](https://github.com/mapbox/mapbox-maps-android/pull/333))
* Fix crash if the belowLayerId doesn't exist on the current style ([#330](https://github.com/mapbox/mapbox-maps-android/pull/330))
* Fix skybox cubemap sampling with metal rendering backend
* Style pack download does not cancel pending tileset descriptor resolving
* Fixed the excessive network usage during map browsing caused by losing of the expiration date and the etag for the cached files
* Fix excessive network usage for delayed tile requests
* On style pack update we reset only glyphs and only when the updated options require less glyphs than currently available and we make sure ambient cache size limit is never exceeded

> * Emit 'StyleDataLoaded' and 'SourceDataLoaded' synchronously if possible, so that developers could modify style and sources before map starts rendering style

Reviewer: ```suggestion
* Emit `StyleDataLoaded` and `SourceDataLoaded` synchronously if possible, so that developers could modify style and sources before map starts rendering style
```

      * Otherwise method will be synchronous.
      *
      * @param value the feature collection
      * @param onDataParsed optional callback notifying when data is parsed on a worker thread
      */
     fun featureCollection(
       value: FeatureCollection,
       onDataParsed: ((GeoJsonSource) -> Unit)? = null
     )
   ```
   Similar functions are introduced for `feature` and `geometry`.

## Features ✨ and improvements 🏁
* Add a `cameraOptions(cameraState, builderBlock)` inline method that helps mutate an existing `CameraState` object. ([#317](https://github.com/mapbox/mapbox-maps-android/pull/317))
* Add selected state handling to annotation plugin ([#316](https://github.com/mapbox/mapbox-maps-android/pull/316))
* Add API for disabling vertical/horizontal scroll gestures ([#319](https://github.com/mapbox/mapbox-maps-android/pull/319))
* Introduce API to enable render cache feature that could bring up rendering performance improvement. ([#326](https://github.com/mapbox/mapbox-maps-android/pull/326))
* Add `removeAnnotationManager` API. ([#330](https://github.com/mapbox/mapbox-maps-android/pull/330))

> * Improve terrain's rendering performance by reusing framebuffer and reducing framebuffer switches
> * Improve terrain's rendering performance by rendering largest drapeable batches first to the render-cache

Reviewer 2: ```suggestion
* Improve terrain's rendering performance 
```

Mapbox welcomes participation and contributions from everyone.

# 10.0.0-beta.20 - May 20, 2021
## Breaking changes ⚠️
* Introduce ResourceOptionsManager to configure the default resource options, and removed the xml configuration options for cache path and tile store path. ([#339](https://github.com/mapbox/mapbox-maps-android/pull/339))
* Rename default ambient cache database to mapbox/maps/ambient_cache.db  ([#314](https://github.com/mapbox/mapbox-maps-android/pull/314))
* Remove the usage of asset path from the codebase, as it is not useful in Android Maps SDK. ([#334](https://github.com/mapbox/mapbox-maps-android/pull/334))
* CameraManager#setBounds return type changed from void to expected
* Expose tileset version for sources that use TileJSON metadata
* ResourceRequest offline-data boolean field is replaced with the source string field, which whether the response came from network, cache or tile store
* Remove GeoJSON specific methods from `StyleManager`. `Map#queryFeatureExtensions` provides the same functionality

> 

Reviewer 3: nit: emtpy line

> 

Reviewer: ```suggestion
```

Mapbox welcomes participation and contributions from everyone.

# 10.0.0-beta.20 - May 20, 2021
## Breaking changes ⚠️
* Introduce ResourceOptionsManager to configure the default resource options, and removed the xml configuration options for cache path and tile store path. ([#339](https://github.com/mapbox/mapbox-maps-android/pull/339))
* Rename ambient cache database to ambient_cache.db  ([#314](https://github.com/mapbox/mapbox-maps-android/pull/314))
* Remove usage of asset path from the codebase, as it is not used in Android SDK. ([#334](https://github.com/mapbox/mapbox-maps-android/pull/334))

> * Parsing geojson on a worker thread.. ([#327](https://github.com/mapbox/mapbox-maps-android/pull/327))

Reviewer 3: I would include more information here in changelog as it is pretty important.

Author: Can you provide detailed information on this? I just copied from #327

* Expose tileset version for sources that use TileJSON metadata
* ResourceRequest offline-data boolean field is replaced with the source string field, which whether the response came from network, cache or tile store
* Remove GeoJSON specific methods from 'StyleManager'. Map#queryFeatureExtensions provides the same functionality

## Features ✨ and improvements 🏁
* Add a `cameraOptions(cameraState, builderBlock)` inline method that helps mutate an existing `CameraState` object. ([#317](https://github.com/mapbox/mapbox-maps-android/pull/317))
* Add selected state handling to annotation plugin ([#316](https://github.com/mapbox/mapbox-maps-android/pull/316))
* Add API for disabling vertical/horizontal scroll gestures ([#319](https://github.com/mapbox/mapbox-maps-android/pull/319))
* Introduce API to enable render cache feature that could bring up rendering performance improvement. ([#326](https://github.com/mapbox/mapbox-maps-android/pull/326))
* Add `removeAnnotationManager` API. ([#330](https://github.com/mapbox/mapbox-maps-android/pull/330))
* Improve terrain's rendering performance by reusing framebuffer and reducing framebuffer switches

> * Improve terrain's rendering performance by rendering largest drapeable batches first to the render-cache

Reviewer 3: Not sure it should be public.
